WeatherForecastApp/Controller/LocationDetailViewController.swift

```diff
@@ -89,29 +89,31 @@ class LocationDetailViewController: UIViewController {
             
             pageViewController.weatherLocations = source.weatherLocations
             pageViewController.setViewControllers([pageViewController.createLocationDetailViewController(forPage: locationIndex)], direction: .forward, animated: false, completion: nil)
-            
         }
     }
+    
     @IBAction func pageControlTapped(_ sender: UIPageControl) {
         let pageViewController = UIApplication.shared.windows.first!.rootViewController as! PageViewController
         
         var direction: UIPageViewController.NavigationDirection = .forward
         if sender.currentPage < locationIndex {
             direction = .reverse
-    }
+        }
         
         pageViewController.setViewControllers([pageViewController.createLocationDetailViewController(forPage: sender.currentPage)], direction: direction, animated: true, completion: nil)
-        
     }
 }
+
 extension LocationDetailViewController: UITableViewDelegate, UITableViewDataSource {
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return weatherDetail.dailyWeatherData.count
+        return weatherDetail?.dailyWeatherData.count ?? 0
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath) as! DailyTableViewCell
```

Comment:
Do not use force unwrapping because if you don't get that cell the app crashes here